symengine/visitor.cpp

```diff
@@ -134,6 +134,53 @@ set_basic free_symbols(const Basic &b)
     return visitor.apply(b);
 }
 
+class FunctionSymbolsVisitor : public BaseVisitor<FunctionSymbolsVisitor>
+{
+public:
+    set_basic s;
+    uset_basic v;
+
+    void bvisit(const Subs &x)
+    {
+        set_basic set_ = function_symbols(*x.get_arg());
+        for (const auto &p : x.get_variables()) {
+            set_.erase(p);
+        }
+        s.insert(set_.begin(), set_.end());
+        for (const auto &p : x.get_point()) {
+            auto iter = v.insert(p->rcp_from_this());
+            if (iter.second) {
+                p->accept(*this);
+            }
+        }
+    }
+
+    void bvisit(const Basic &x)
+    {
+        if (is_a<FunctionSymbol>(x)) {
+            s.insert(x.rcp_from_this());
+        }
```

Comment:
Can you remove the above 3 lines and add a new method,
```
void bvisit(const FunctionSymbo &x) {
    s.insert(x.rcp_from_this());
}
```

Reply:
In order to visit also the arguments of the FunctionSymbol one needs the method below. But then the lower part of the method is copied from `void bvisit(const Basic &x)`. Which version do you prefer, the original one or the one below?
```    
    void bvisit(const FunctionSymbol &x) {
        s.insert(x.rcp_from_this());
        
        for (const auto &p : x.get_args()) {
            auto iter = v.insert(p->rcp_from_this());
            if (iter.second) {
                p->accept(*this);
            }
        }
    }
```

Reply:
@isuruf May I ask for an answer. I'm not sure which version you prefer.